cabal-validate/src/Step.hs

```diff
@@ -0,0 +1,63 @@
+-- | The steps that can be run by @cabal-validate@.
+
+module Step
+  ( Step (..)
+  , displayStep
+  , nameToStep
+  , parseStep
+  ) where
+
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+
+-- | A step to be run by @cabal-validate@.
+data Step
+  = PrintConfig
+  | PrintToolVersions
+  | Build
+  | Doctest
+  | LibTests
+  | LibSuite
+  | LibSuiteExtras
+  | CliTests
+  | CliSuite
+  | SolverBenchmarksTests
+  | SolverBenchmarksRun
+  | TimeSummary
+  deriving (Eq, Enum, Bounded, Show)
```

Comment:
Do we really need Show? Given that there’s no Read and that there’s a dedicated display function.

Reply:
Ability to debug is nice.